src/SonarQube.SonarLint.Runner/Configuration.cs

```
@@ -29,66 +29,39 @@
 using SonarLint.Common;
 using System.Reflection;
 using SonarLint.Rules.CSharp;
+using SonarLint.Helpers;
 
 namespace SonarLint.Runner
 {
     public class Configuration
     {
-        private class RuleParameterValue
-        {
-            public string ParameterKey { get; set; }
-            public string ParameterValue { get; set; }
-        }
-        private class RuleParameterValues
-        {
-            public string RuleId { get; set; }
-            public List<RuleParameterValue> ParameterValues { get; set; } = new List<RuleParameterValue>();
-        }
-
         private readonly ImmutableArray<DiagnosticAnalyzer> nonTemplateAnalyzers;
-        private readonly IImmutableList<RuleParameterValues> parameters;
         private readonly AnalyzerLanguage language;
+        private readonly XDocument xml;
+
+        public string Path { get; private set; }
 
         public bool IgnoreHeaderComments { get; }
         public IImmutableList<string> Files { get; }
         public IImmutableSet<string> AnalyzerIds { get; }
 
-        public Configuration(XContainer xml, AnalyzerLanguage language)
+        public Configuration(string path, AnalyzerLanguage language)
         {
+            Path = path;
```

@tvsonar Here we should fail fast is the provided XML filename doesn't match `SonarLint.xml`